generator_process/actions/prompt_to_image.py

```diff
@@ -236,6 +236,25 @@ def approximate_decoded_latents(latents):
 
     return latents_ubyte.numpy()
 
+def model_snapshot_folder(model, preferred_revision: str | None = None):
+    """ Try to find the preferred revision, but fallback to another revision if necessary. """
+    storage_folder = model
+    if os.path.exists(os.path.join(storage_folder, 'model_index.json')): # converted model
+        snapshot_folder = storage_folder
+    else: # hub model
+        ref_path = None
+        for revision in os.listdir(os.path.join(storage_folder, "refs")):
+            ref_path = os.path.join(storage_folder, "refs", revision)
+            if revision == preferred_revision:
+                break
+        if ref_path is None:
+            return None
+        with open(ref_path) as f:
+            commit_hash = f.read()
+
+        snapshot_folder = os.path.join(storage_folder, "snapshots", commit_hash)
+    return snapshot_folder
```

Comment:
```suggestion
        if len(os.listdir(snapshot_folder)) <= 1:
            raise ValueError(f"revision \"{revision}\" is not installed for model {model}")
    return snapshot_folder
```
Downloading the fp16 branch also downloads `model_index.json` for the main branch. This leads to the function returning the partially downloaded main snapshot when someone tries to disable half precision while using the model. It'll later error with `OSError: Error no file named diffusion_pytorch_model.bin` which is ambiguous with an incorrectly downloaded model.